src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -993,7 +1021,60 @@ public function get_current_depth(): int {
 	 * @return bool Whether an element was found.
 	 */
 	private function step_initial(): bool {
-		$this->bail( 'No support for parsing in the ' . WP_HTML_Processor_State::INSERTION_MODE_INITIAL . ' state.' );
+		$token_name = $this->get_token_name();
+		$token_type = $this->get_token_type();
+		$op_sigil   = '#tag' === $token_type ? ( parent::is_tag_closer() ? '-' : '+' ) : '';
+		$op         = "{$op_sigil}{$token_name}";
+
+		/*
+		 * > A character token that is one of U+0009 CHARACTER TABULATION,
+		 * > U+000A LINE FEED (LF), U+000C FORM FEED (FF),
+		 * > U+000D CARRIAGE RETURN (CR), or U+0020 SPACE
+		 *
+		 * Parse error: ignore the token.
+		 */
+		if ( '#text' === $op ) {
+			$text = $this->get_modifiable_text();
+			if ( strlen( $text ) === strspn( $text, " \t\n\f\r" ) ) {
+				return $this->step();
+			}
+		}
+
+		switch ( $op ) {
+			/*
+			 * > A comment token
+			 */
+			case '#comment':
+			case '#funky-comment':
+			case '#presumptuous-tag':
+				$this->insert_html_element( $this->state->current_token );
+				return true;
+
+			/*
+			 * > A DOCTYPE token
+			 */
+			case 'html':
+				$contents = $this->get_modifiable_text();
+				if ( ' html' !== $contents ) {
+					/*
+					 * @todo When the HTML Tag Processor fully parses the DOCTYPE declaration,
+					 *       this code should examine the contents to set the compatability mode.
+					 */
+					$this->bail( 'Cannot process any DOCTYPE other than a normative HTML5 doctype.' );
+				}
+
+				/*
+				 * > Then, switch the insertion mode to "before html".
+				 */
+				$this->state->insertion_mode = WP_HTML_Processor_State::INSERTION_MODE_BEFORE_HTML;
+				return true;
```

Comment:
thanks. interestingly enough this is because we don't insert a node for the DOCTYPE declaration.

I'm not sure what to do either, because it's not accessible and it's not a child or parent of the root HTML node.

Reply:
What do you mean by "it's not accessible?"

My initial thoughts are that this is an important part of the document and it should be reached via an interface like `next_token`.